File: src/main/java/ru/igrey/dev/service/CasService.java
package ru.igrey.dev.service;

import com.google.gson.Gson;
import org.apache.http.HttpEntity;
import org.apache.http.client.methods.CloseableHttpResponse;
import org.apache.http.client.methods.HttpGet;
import org.apache.http.impl.client.CloseableHttpClient;
import org.apache.http.impl.client.HttpClients;
import org.apache.http.util.EntityUtils;
import ru.igrey.dev.domain.Cas;
import ru.igrey.dev.domain.CasUsers;

import java.util.Set;
import java.util.stream.Collectors;

/**
 * Created by sanasov on 10.04.2017.
 */
public class CasService {

    public CasUsers casUsers(Set<Long> casIds) {
        return new CasUsers(casIds.stream()
                .map(this::casUser)
                .collect(Collectors.toList())
        );
    }

    public Cas casUser(Long casId) {
        HttpGet request = new HttpGet("https://domclick.ru/cas/rest/v1/srv/users/" + casId);
        initHeader(request);
        try (CloseableHttpClient httpClient = HttpClients.createDefault();
             CloseableHttpResponse response = httpClient.execute(request)) {

            HttpEntity entity = response.getEntity();
            if (entity != null) {
                String json = EntityUtils.toString(entity);
                return new Gson().fromJson(json, Cas.class);
            }
        } catch (Exception e) {
            throw new RuntimeException("Could not get Cas by casId: " + casId);
        }
        return null;
    }

    private void initHeader(HttpGet request) {
        request.setHeader("Content-Type", "application/json");
        request.setHeader("Accept", "application/json");
    }
}

File: src/main/kotlin/ru/igrey/dev/service/CasService.kt
package ru.igrey.dev.service

import com.google.gson.Gson
import lombok.extern.slf4j.Slf4j
import mu.KLogging
import org.apache.http.client.methods.HttpGet
import org.apache.http.impl.client.HttpClients
import org.apache.http.util.EntityUtils
import ru.igrey.dev.config.HttpClientConfig.CAS_TOKEN
import ru.igrey.dev.config.HttpClientConfig.CAS_URI
import ru.igrey.dev.domain.Cas
import ru.igrey.dev.domain.CasUsers

@Slf4j
class CasService {
    fun casUserInfo(phoneOrCasId: String): Cas? {
        var phone = phoneOrCasId
        if (phoneOrCasId.length.toLong() == 11L) {
            phone = phoneOrCasId.substring(1)
        }
        return if (phoneOrCasId.length < 10L) casUser(phoneOrCasId.toLong())
        else casUserByPhone(phone)
    }

    fun casUsers(casIds: Set<Long>) = CasUsers(casIds.mapNotNull { casUser(it) })

    fun casUser(casId: Long?): Cas? {
        val request = HttpGet("$CAS_URI/v1/srv/users/" + casId!!)
        initHeader(request)
        try {
            HttpClients.createDefault().use { httpClient ->
                httpClient.execute(request).use { response ->

                    val entity = response.entity
                    if (entity != null) {
                        val json = EntityUtils.toString(entity)
                        return Gson().fromJson(json, Cas::class.java)
                    }
                }
            }
        } catch (e: Exception) {
            logger.error("Error casUser " + casId + "\n" + e.message)
            return null
        }

        return null
    }

    private fun casUserByPhone(phone: String): Cas? {
        val request = HttpGet("$CAS_URI/v1/srv/user?login=$phone")
        initHeader(request)
        try {
            HttpClients.createDefault().use { httpClient ->
                httpClient.execute(request).use { response ->

                    val entity = response.entity
                    if (entity != null) {
                        var json = EntityUtils.toString(entity)
                        json = Gson().toJson(Gson().fromJson(json, Map::class.java)["data"])
                        return Gson().fromJson(json, Cas::class.java)
                    }
                }
            }
        } catch (e: Exception) {
            logger.error("Error casUserByPhone " + phone + "\n" + e.message)
            return null
        }

        return null
    }

    private fun initHeader(request: HttpGet) {
        request.setHeader("Content-Type", "application/json")
        request.setHeader("Accept", "application/json")
        request.setHeader("Authorization", CAS_TOKEN)
    }

    companion object : KLogging()
}